File: software/mobile/android/core-model/src/main/kotlin/com/avlyubimov/svc/core/model/DeviceRepository.kt
package com.avlyubimov.svc.core.model

import kotlinx.coroutines.flow.StateFlow

enum class ConnectionState {
    DISCONNECTED,
    SCANNING,
    CONNECTED,
}

data class DiscoveredDevice(
    val id: String,
    val name: String,
    val signalStrength: Int,
)

data class DeviceSnapshot(
    val connectionState: ConnectionState,
    val telemetry: TelemetrySnapshot,
    val events: List<String>,
)

interface DeviceRepository {
    val snapshot: StateFlow<DeviceSnapshot>
    suspend fun discover(): List<DiscoveredDevice>
    suspend fun connect(id: String)
    suspend fun refresh()
}

File: software/mobile/ios/SVCMobile/Sources/Data/DeviceRepository.swift
import Foundation

enum ConnectionState: String {
    case disconnected
    case scanning
    case connected
}

struct DiscoveredDevice: Identifiable, Equatable {
    let id: UUID
    let name: String
    let signalStrength: Int
}

struct DeviceSnapshot {
    let connectionState: ConnectionState
    let telemetry: TelemetrySnapshot
    let events: [String]
}

protocol DeviceRepository {
    func discover() -> [DiscoveredDevice]
    func connect(to id: UUID)
    func currentSnapshot() -> DeviceSnapshot
}
